smolyay/basis.py

```diff
@@ -328,141 +296,151 @@ def derivative(self, x):
         Returns
         -------
         float
-            Value of the derivative of Chebyshev polynomials of first kind.
+            Value of the derivative of Chebyshev polynomials of second kind.
 
         Raises
         ------
         ValueError
-            if input is outside the domain [-1,1].
+            if input is outside the domain [-1, 1].
         """
-        if numpy.any(numpy.greater(x, 1)) or numpy.any(numpy.less(x, -1)):
-            raise ValueError("Input is outside the domain [-1,1]")
-        if self._derivative_polynomial is None:
-            self._derivative_polynomial = ChebyshevFirstKind(self._n+1)
         x = numpy.asarray(x)
         y = numpy.zeros(x.shape)
-        flag2 = x == 1
-        flag3 = x == -1
-        flag1 = ~(flag2 | flag3)
-        y[flag1] =  ((self._n+1)*self._derivative_polynomial(x[flag1]) -
-                     x[flag1]*self(x[flag1]))/(x[flag1]**2-1)
-        y[flag2] = self._n*(self._n + 1)*(self._n + 2)/3
-        y[flag3] = ((-1)**(self._n+1))*self._n*(self._n + 1)*(self._n + 2)/3
+        u_limit = self.n * (self.n + 1) * (self.n + 2) / 3
+        flag_upper = x == 1
+        y[flag_upper] = u_limit
+        
+        flag_lower = x == -1
+        y[flag_lower] = (-1) ** (self.n+1) * u_limit
+
+        flag = ~(flag_upper | flag_lower)
+        y[flag] = (
+            (self.n + 1) * scipy.special.eval_chebyt(self.n + 1, x[flag])
+            - x[flag] * scipy.special.eval_chebyu(self.n, x[flag])
+        ) / (x[flag] ** 2 - 1)
         if y.ndim == 0:
             y = y.item()
         return y
 
-    @classmethod
-    def make_nested_set(cls, exactness):
-        """Create a nested set of Chebyshev polynomials.
 
-        A nested set is created up to a given level of ``exactness``,
-        which corresponds to a highest-order Chebyshev polynomial of
-        degree ``n = 2**(exactness + 1) - 1``.
+class Trigonometric(BasisFunction):
+    r"""Trigonometric basis functions.
```

Comment:
I would prefer `n`, whatever it represents, to correspond with the parameter used by the TrigonometricPointSet, but I want @mzf0069 to weigh in before doing anything or updating the documentation.

Reply:
They do not necessarily need to be the same because they are different objects

Reply:
> Please update the documentation for this class, it is out of date after the changes made to remove sigma.
> 
> Also, what would be the appropriate new name for `n`? It could also be `degree`, if you add some documentation explaining that:
> ```
> e^{i n x} = (e^{ix})^n
> ```
> It could also be called `frequency` or something like that, but then there is the weirdness of it being allowed to be a negative number (which frequencies usually are not).
> 
> Maybe consult with @mzf0069 on these changes

I would rename it to `node` or `node_frequency`. Each frequency has two nodes (one negative and the other positive --- except the first one).

Reply:
Or `wavenumber`?

Reply:
> Hmm, interesting. Including `node` works nicely for interpolation, but this is a more general concept of a basis function. I would use just `frequency`. Negative frequencies are fine mathematically:
> https://en.wikipedia.org/wiki/Negative_frequency

If negative frequency are fine, I would probably stick with `frequency` since the phrase is more common in the trig sparse grid papers.

Reply:
Okay, I've made the changes. Frequency can be any integer and I've adjusted the call and the tests to accommodate the negative frequencies.